service/worker/deployment/workflow.go

```diff
@@ -0,0 +1,174 @@
+		metrics    sdkclient.MetricsHandler
+	}
+)
+
+var (
+	defaultActivityOptions = workflow.ActivityOptions{
+		ScheduleToCloseTimeout: 1 * time.Hour,
+		StartToCloseTimeout:    30 * time.Second,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval: 1 * time.Second,
+			MaximumInterval: 60 * time.Second,
+		},
+	}
+)
+
+const continueAsNewThreshold = 1000
+
+func DeploymentWorkflow(ctx workflow.Context, deploymentWorkflowArgs Deployment) error {
+	deploymentWorkflowRunner := &DeploymentWorkflowRunner{
+		deployment: deploymentWorkflowArgs,
+		ctx:        ctx,
+		logger:     sdklog.With(workflow.GetLogger(ctx), "wf-namespace", deploymentWorkflowArgs.Namespace),
+		metrics:    workflow.GetMetricsHandler(ctx).WithTags(map[string]string{"namespace": deploymentWorkflowArgs.Namespace}),
+	}
+	// TQ should be set to nil here since signal handler updates it
+	deploymentWorkflowRunner.deployment.TaskQueue = nil
+	return deploymentWorkflowRunner.run()
+}
+
+func (d *DeploymentWorkflowRunner) run() error {
+
+	// Query Handler
+	err := workflow.SetQueryHandler(d.ctx, "list-task-queues", func(input []byte) ([]*TaskQueue, error) {
+		return d.deployment.TaskQueue, nil
+	})
+	if err != nil {
+		d.logger.Info("SetQueryHandler failed for list-task-queue with the error: " + err.Error())
+		return err
+	}
+
+	updateDeploymentSignalChannel := workflow.GetSignalChannel(d.ctx, "update_deployment")
+	updateBuildIDSignalChannel := workflow.GetSignalChannel(d.ctx, "update_task_queue_build_id")
+	var signalCount int
+
+	selector := workflow.NewSelector(d.ctx)
+	selector.AddReceive(updateDeploymentSignalChannel, func(c workflow.ReceiveChannel, more bool) {
+		signalCount++
+		var signalTaskQueue TaskQueue
+		updateDeploymentSignalChannel.Receive(d.ctx, &signalTaskQueue)
+		d.logger.Info("Received Signal to update Deployment!", "signal", signalTaskQueue)
+
+		// Process Signal
+		d.deployment.TaskQueue = append(d.deployment.TaskQueue, &signalTaskQueue)
+
+		// TODO Shivam: Start an entity workflow for DeploymentName:
+
+	})
+	selector.AddReceive(updateBuildIDSignalChannel, func(c workflow.ReceiveChannel, more bool) {
+		signalCount++
+		// Process Signal
+
+		var new_buildID string
+		updateBuildIDSignalChannel.Receive(d.ctx, &new_buildID)
+
+		ctx := workflow.WithActivityOptions(d.ctx, defaultActivityOptions)
+		var updatedTaskQueues []*TaskQueue
+		err := workflow.ExecuteActivity(ctx, d.a.VerifyTaskQueueDefaultBuildID, d.deployment, d.deployment.BuildID).Get(d.ctx, &updatedTaskQueues)
+		if err != nil {
+			d.logger.Error("Task Queue default buildID verification activity failed with error : %s", err.Error())
+			return
+		}
+		// update taskQueue list for the deployment as some task queues might have moved to a different buildID
+		d.deployment.TaskQueue = updatedTaskQueues
+
+		// Updating buildID of the task queues in consideration
+		var res bool
+		err = workflow.ExecuteActivity(ctx, d.a.UpdateTaskQueueDefaultBuildID, d.deployment, new_buildID).Get(d.ctx, &res)
```

Comment:
If this (or the other activity) is expected to be very fast (e.g. a single rpc), then you can make it a local activity instead. But you don't have to worry about that right now